frontend/app/login/components/LoginForm.tsx

```diff
@@ -2,17 +2,80 @@
 
 import { Button } from 'components/buttons/Button'
 import { Input } from 'components/forms/Input'
+import { useRouter } from 'next/navigation'
+import { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import useLogin from '../hooks/useLogin'
+
+type FormType = {
+  username: string
+  password: string
+}
 
 export const LoginForm: React.FC = () => {
-  const handleOnSubmit = () => {
-    console.log('SUBMITTED!')
+  const router = useRouter()
+  const { mutate, isLoading } = useLogin()
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isSubmitting }
+  } = useForm<FormType>()
+  const [isError, setIsError] = useState(false)
+
+  const onSubmit = async (formdata: FormType): Promise<void> => {
+    setIsError(false)
+    mutate(formdata, {
+      onSuccess: () => {
+        // TODO: Save token data to cookies
+        router.push('/')
+      },
+      onError: () => setIsError(true)
+    })
+  }
+
+  const errorFieldMessage = (error?: string): React.ReactNode => {
+    return (
+      !!error && (
+        <div className="px-[0.75rem] pt-1" role="presentation" aria-label="error-field-message">
+          <p className="text-sm text-red-500">{error}</p>
+        </div>
+      )
+    )
   }
 
   return (
-    <form className="flex w-full flex-col gap-4" onSubmit={handleOnSubmit}>
-      <Input label="Username" />
-      <Input label="Password" type="password" />
-      <Button label="Sign In" />
+    <form className="flex w-full flex-col gap-4" onSubmit={handleSubmit(onSubmit)}>
+      {isError && (
+        <div
+          className="rounded-lg bg-red-500 px-2 py-3"
+          role="presentation"
+          aria-label="login-error"
+        >
+          <p className="text-center text-sm text-white">
+            Sorry, we can&apos;t find an account with this username. Please try again.
+          </p>
+        </div>
+      )}
+      <div>
+        <Input
+          label="Username"
+          isError={!!errors.username?.message}
+          {...register('username', { required: 'Username is required.' })}
```

Comment:
This might be outside the scope of what you're doing - but have you ever tried the i18n of react? It's really nifty. Removes the hard-coded error languages from your markup and into a configuration file